### PR TITLE
[Play] - Base app set up with TypeScript and Storybook

### PR DESCRIPTION
**Pull request by drewjhart**
This PR sets up the base layer of the ReactJS student app (tentatively called Play). Resolves https://github.com/rightoneducation/righton-app/issues/582

This provides the following:
1. ReactJS app using Yarn, with TypeScript
2. Container layer `GameSessionContainer.tsx` that imports MockGameSession.JSON and parses it using the parser from `@righton/networking`
3. Base component 'GameInProgress.tsx' that will be built upon in subsequent PR's 
4. Preliminary storybook for `GameInProgress`

Additionally, this updates the `manifest.json` on all app to include an exclamation mark on RightOn and it also exports the `GameSessionfromAWSGameSession` parser and associated types from `@righton/networking` for use with the mock data.

This also includes merged PR's as below:
[[Play] - GameInProgress UI Framework #577:](https://github.com/rightoneducation/righton-app/pull/577)
Lays out the rough `div` structure for the UI framework per Figma

[[Play] - HeaderContent and Timer Components #578:](https://github.com/rightoneducation/righton-app/pull/578)
Adds components for `HeaderContent` and `Timer`

[[Play] - Footer Content component #579:](https://github.com/rightoneducation/righton-app/pull/579)
Adds components for `FooterContent`

### CODE REVIEW

**Comment by drewjhart** [comment]
Hi @maniramezan, I've added you to this review to verify if you've got any concerns. This handles the base project setup and also has integrated some smaller PR's containing common components. Ideally, I'd like to get approval on [PR580](https://github.com/rightoneducation/righton-app/pull/580), merge it into this and then merge this into `dev`. From there, I'd like to work on individual pages (and PR any new components similar to how we are doing this). I'm concerned that if we don't merge this after PR580, it will bloat pretty hard.

File: play/package-lock.json
Comment (maniramezan) [inline play/package-lock.json:1]: Can be removed if you're using `yarn`

File: play/package.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      "not dead",
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
```
Comment (maniramezan) [inline play/package.json:62]: Commented in the other PR, RE creating linter and prettifier. This PR is a good place for that or can be a PR on top of this one

File: play/public/icon.png
Comment (maniramezan) [inline play/public/icon.png:1]: Intentionally a png? I believe they don't scale well in web, but again not a web expert and has been away from it so not sure what's the common practice these days

File: play/src/img/MonsterIcon1.png
Comment (maniramezan) [inline play/src/img/MonsterIcon1.png:1]: Same comment for these assets here RE vector vs. pixel base assets.

File: play/src/react-app-env.d.ts
Comment (maniramezan) [inline play/src/react-app-env.d.ts:1]: hmm, is this file needed?

File: play/src/components/ScoreIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        { (newPoints && newPoints > 0) ? 
+          <div className={classes.newPointsContainer}>
+            <Typography className={classes.text}> {`+${newPoints}`} </Typography>
+          </div>
+        : 
+          null
+        }
+      </div>
+      <div className={classes.scoreContainer}> 
+        <Typography className={classes.text}> {isNullOrUndefined(currentScore) ? 0 : currentScore} </Typography>
```
Comment (maniramezan) [inline play/src/components/ScoreIndicator.tsx:42]: Intentional spaces in the beginning and end?

File: play/src/components/ScoreIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import { useState, useEffect } from 'react';
+import { makeStyles, ThemeProvider} from '@mui/styles';
+import { Typography } from "@mui/material";
+import { isNullOrUndefined } from '@righton/networking';
+
+interface ScoreIndicatorProps {
+  newPoints: number;
+  score: number | null;
```
Comment (maniramezan) [inline play/src/components/ScoreIndicator.tsx:8]: Any reason we go for nullable score? We should try to clean these ones up as well and have them defaulted to 0 always to reduce null checks and increase code readability

File: play/package-lock.json
Comment (drewjhart) [reply play/package-lock.json:1]: resolved

File: play/package.json (hunk 1 above)
Comment (drewjhart) [reply play/package.json:65]: ESLint and Prettier have been added to `play`. Code has been formatted using prettier and linting errors have been corrected for. ESLint configuration is currently using airbnb standards and prettier is all default except we are going with single quotes (if you have any insight on rules here, just let me know). I'll add more detail about this in the PR description above.

File: play/src/react-app-env.d.ts
Comment (drewjhart) [reply play/src/react-app-env.d.ts:1]: This comes from `create-react-app` and references type declarations that allow for importing of images:

https://github.com/facebook/create-react-app/issues/6560#issuecomment-792323361
https://stackoverflow.com/questions/67262914/what-is-the-react-app-env-d-ts-in-a-react-typescript-project-for

Kind of an annoying file

File: play/src/components/ScoreIndicator.tsx (hunk 2 above)
Comment (drewjhart) [reply play/src/components/ScoreIndicator.tsx]: removed

File: play/src/img/MonsterIcon1.png
Comment (drewjhart) [reply play/src/img/MonsterIcon1.png:1]: images replaced with svg

File: play/public/icon.png
Comment (drewjhart) [reply play/public/icon.png:1]: image replaced with svg

File: play/src/components/ScoreIndicator.tsx (hunk 3 above)
Comment (drewjhart) [reply play/src/components/ScoreIndicator.tsx:51]: yeah, totally agreed (see comment in PR580, cardanswers). An issue has been added to github for this as I'd like to handle it in a subsequent pr (and kill off some UI stuff first)
